lute/cli/include/lute/reporter.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <cstdio>
+#include <string>
+#include <vector>
+
+// Interface for reporting output and errors from CLI operations
+class LuteReporter
+{
+public:
+    virtual ~LuteReporter() = default;
+
+    // Report an error message (typically to stderr)
+    virtual void reportError(const std::string& message) = 0;
+
+    // Report normal output (typically to stdout)
+    virtual void reportOutput(const std::string& message) = 0;
+
+    // Variadic template version for formatted error messages using printf-style format specifiers
+    template<typename... Args>
+    void formatError(const char* format, Args&&... args)
+    {
+        reportError(formatMessage(format, std::forward<Args>(args)...));
+    }
+
+    // Variadic template version for formatted output messages using printf-style format specifiers
+    template<typename... Args>
+    void formatOutput(const char* format, Args&&... args)
+    {
+        reportOutput(formatMessage(format, std::forward<Args>(args)...));
+    }
+
+private:
+    // Format implementation using snprintf
+    template<typename... Args>
+    static std::string formatMessage(const char* format, Args&&... args)
+    {
+        // Calculate required buffer size
+        int size = std::snprintf(nullptr, 0, format, std::forward<Args>(args)...);
+
+        if (size > 0)
+        {
+            // Allocate buffer and format the string
+            std::vector<char> buffer(size + 1);
+            std::snprintf(buffer.data(), buffer.size(), format, std::forward<Args>(args)...);
```

Comment:
```suggestion
            std::snprintf(buffer.data(), buffer.size(), format, args...);
```
I think forwarding twice isn't what we want here, plus `snprintf` won't really benefit from move semantics.